### PR TITLE
[Java] Enable GCS server when running java unit tests 

### PR DESCRIPTION
**Pull request by ffbin**
## Why are these changes needed?
Enable gcs service when run java testcase.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.

### CODE REVIEW

File: java/runtime/src/main/java/org/ray/runtime/runner/RunManager.java [hunk 1]
```diff
@@ -182,7 +182,7 @@ public void startRayProcesses(boolean isHead) {
     }
   }
 
-  private void startRedisServer() {
+  private void startGcs() {
```
Comment (jovany-wang) [inline java/runtime/src/main/java/org/ray/runtime/runner/RunManager.java]: Better to separate 2 methods:
1. startRedisServer
2. startGcsServer

File: python/ray/services.py [hunk 2]
```diff
@@ -54,6 +54,8 @@
     "core/src/ray/raylet/raylet_monitor")
 RAYLET_EXECUTABLE = os.path.join(
     os.path.abspath(os.path.dirname(__file__)), "core/src/ray/raylet/raylet")
+GCS_SERVER_EXECUTABLE = os.path.join(
```
Comment (jovany-wang) [inline python/ray/services.py]: We should copy the `gcsserver` binary to wheel in `setup.py`

**Review by jovany-wang — CHANGES_REQUESTED** [review]

File: .travis.yml [hunk 3]
```diff
@@ -64,6 +63,7 @@ matrix:
       script:
         - ./ci/suppress_output bash src/ray/test/run_core_worker_tests.sh
         - ./ci/suppress_output bash streaming/src/test/run_streaming_queue_test.sh
+        - ./java/test.sh
```
[the remark this answers is not in the document]
Comment (ffbin) [reply .travis.yml]: we have add a separated CI job: 
    - os: linux
      env:
        - TESTSUITE=gcs_service
        - JDK='Oracle JDK 8'
        - RAY_GCS_SERVICE_ENABLED=true
        - PYTHON=3.5 PYTHONWARNINGS=ignore
        - RAY_INSTALL_JAVA=1
      install:
        - eval `python $TRAVIS_BUILD_DIR/ci/travis/determine_tests_to_run.py`
        - ./ci/travis/install-bazel.sh
        - ./ci/suppress_output ./ci/travis/install-dependencies.sh
        - export PATH="$HOME/miniconda/bin:$PATH"
        - ./ci/suppress_output ./ci/travis/install-ray.sh
      script:
        - ./ci/suppress_output bash src/ray/test/run_core_worker_tests.sh
        - ./ci/suppress_output bash streaming/src/test/run_streaming_queue_test.sh
        - ./java/test.sh

File: python/ray/services.py (hunk 2 above)
Comment (ffbin) [reply python/ray/services.py]: We have copy gcsserver in setup.py now.

File: java/runtime/src/main/java/org/ray/runtime/runner/RunManager.java [hunk 4]
```diff
@@ -205,6 +205,28 @@ private void startRedisServer() {
         client.rpush("RedisShards", shard);
       }
     }
+
+    // start gcs server
+    if (System.getenv("RAY_GCS_SERVICE_ENABLED") != null) {
```
[the remark this answers is not in the document]
Comment (ffbin) [reply java/runtime/src/main/java/org/ray/runtime/runner/RunManager.java]: GCS Server is a part of GCS Service(which also has GCS Client)，GCS Client will also check this flag, so RAY_GCS_SERVER_ENABLED is better.

File: java/runtime/src/main/java/org/ray/runtime/runner/RunManager.java (hunk 1 above)
Comment (ffbin) [reply java/runtime/src/main/java/org/ray/runtime/runner/RunManager.java]: Gcs contain Redis and Gcs Server, so start them in one method is ok now.

**Review by jovany-wang — APPROVED** [review]
LGTM

**Pull request merged**